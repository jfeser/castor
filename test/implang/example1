select([p_counter, c_counter], filter(c_id = id_c && p_id = id_p,
alist(filter(succ > counter + 1, log) as lp,
atuple([ascalar(lp.id as p_id), ascalar(lp.counter as p_counter),
alist(filter(lp.counter < counter && counter < lp.succ, log) as lc,
atuple([ascalar(lc.id as c_id), ascalar(lc.counter as c_counter)], cross))], cross))))
