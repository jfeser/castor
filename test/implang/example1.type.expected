(FuncT
 (((FuncT
    (((ListT
       ((TupleT
         (((IntT ((range (Interval 1 1)))) (IntT ((range (Interval 1 4))))
           (ListT
            ((TupleT
              (((IntT ((range (Interval 2 3))))
                (IntT ((range (Interval 2 5)))))
               ((kind Cross))))
             ((count (Interval 1 2))))))
          ((kind Cross))))
        ((count (Interval 2 2))))))
     Child_sum)))
  (Width 2)))